ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlUncaughtExceptionHandler.java

```diff
@@ -32,6 +33,9 @@ public KsqlUncaughtExceptionHandler(final Runnable flusher) {
 
   @SuppressFBWarnings
   public void uncaughtException(final Thread t, final Throwable e) {
+    if (t instanceof StreamThread) {
+      return;
```

Comment:
we might want to create a testing-only constructor here where we can pipe down an action for this if branch. That way we can set a countdown latch or something and make sure that this branch is getting hit instead of relying on sleeps, which both will increase our test run time by a minute and also give us worse coverage

Reply:
@agavra I updated the test so that we keep trying to assert that a specific error message that's passed in through the uncaught handler constructor is printed to stderr, which is only done in this logic branch.